sdk/core/azure-core/src/datetime.cpp

```diff
@@ -342,7 +342,10 @@ T ParseNumber(
   }
 
   ThrowParseError(description);
-  return T();
+
+  // ThrowParseError() will always throw, but there's no way to tell that to compiler, so a return
+  // statement is required. It is not possible to cover the return line with tests.
+  return T(); // LCOV_EXCL_LINE
```

Comment:
👍

Is this the type of place we'd use `AzureNoReturnPath`?

Reply:
Maybe, the `_azure_UNREACHABLE_CODE()`. But I'd do it in a separate PR.

Reply:
#3185